terminus/generators/opendrive_generator.py

```diff
@@ -0,0 +1,133 @@
+from file_generator import FileGenerator
+from geometry.point import Point
+from geometry.latlon import LatLon
+from city_visitor import CityVisitor
+from opendrive_id_mapper import OpenDriveIdMapper
+import math
+
+
+class OpenDriveGenerator(FileGenerator):
+
+    def __init__(self, city, origin):
+        super(OpenDriveGenerator, self).__init__(city)
+        self.origin = origin
+        self.id_mapper = OpenDriveIdMapper(city)
+
+    # # Start/End document handling
+    def start_document(self):
+        self.id_mapper.run()
+
+    def end_city(self, city):
+        print "Finishing doc"
+        self._wrap_document_with_contents_for(city)
+
+    def start_street(self, street):
+        self.start_road(street)
+
+    def start_trunk(self, street):
+        self.start_road(street)
+
+    def start_road(self, road):
+        road_id = self.id_for(road)
+        road_content = self._contents_for(road, segment_id=road_id)
+        self._append_to_document(road_content)
+
+    def id_for(self, object):
+        return self.id_mapper.id_for(object)
+
+    def city_template(self):
+        return """
+        <?xml version="1.0" standalone="yes"?>
+          <OpenDRIVE xmlns="http://www.opendrive.org">
+            <header revMajor="1" revMinor="1" name="{{model.name}}" version="1.00" north="0.0000000000000000e+00" south="0.0000000000000000e+00" east="0.0000000000000000e+00" west="0.0000000000000000e+00" maxRoad="{{model.roads_count()}}" maxJunc="0" maxPrg="0">
+            </header>
+            {{inner_contents}}
+        </OpenDRIVE>"""
+
+    def street_template(self):
```

Comment:
Very minor, but can we have a `road_template` method and let `street_template` and `trunk_template` call it?